src/DependencyInjection/DbToolsExtension.php

```diff
@@ -37,10 +37,12 @@ public function load(array $configs, ContainerBuilder $container): void
         $container->setParameter('db_tools.backupper.options', $config['backupper_options']);
         $container->setParameter('db_tools.backup_expiration_age', $config['backup_expiration_age']);
         $container->setParameter('db_tools.excluded_tables', $config['excluded_tables'] ?? []);
+        $container->setParameter('db_tools.backup_timeout', $config['backup_timeout']);
```

Comment:
May be values should be validated here (not negative or not zero for example).

Reply:
The symfony process also validates the timeout. I would explicitly allow 0 to let a process run indefinitely if the user so wishes.

```
    /**
     * Validates and returns the filtered timeout.
     *
     * @throws InvalidArgumentException if the given timeout is a negative number
     */
    private function validateTimeout(?float $timeout): ?float
    {
        $timeout = (float) $timeout;

        if (0.0 === $timeout) {
            $timeout = null;
        } elseif ($timeout < 0) {
            throw new InvalidArgumentException('The timeout value must be a valid positive integer or float number.');
        }

        return $timeout;
    }
```

Reply:
OK, I have numerous comments about this piece of code:

- First, it should be a `int`, not a `float`.
- `float` comparison is dangerous, because of float imprecision, so you should not check against 0.0, but instead `if (null === $timeout) { /* ... */ } else if ($timeout <= 0) { throw ... }`

Reply:
Hello again, this is not my code, this is code from the symfony repository: https://github.com/symfony/process/blob/820da2d703f2904ae04bbee634af705febfc1d68/Process.php#L1351
i'm not gonna take it upon me to change things there

i just copied this code into this comment to tell you that it is validated by the process itself, so i dont see why we should validate it to be a certain way as well, but if you want so i can add some validation to your repo for the timeout as well.

I however disagree with the timeout needing to be bigger than 0, because in pretty much any configuration you can set timeouts to 0 for things you want to run indefinitely (e.g. php's set_time_limit)

Reply:
Yes that's true, but I do prefer things to be explicit, i.e. `null` means no timeout, a value means there is one, it's pretty much the convention we took everywhere else in this project (there might be exceptions thought). @SimonMellerin what's your opinion ?

Reply:
so "null" as a string should lead to no timeout? because no value (so "null" for real) in the config currently leads to the default 600 which is the hardcoded timeout right now. I would leave it at "600 if no value provided" for backwards compatibility reasons

Reply:
Validation and string to float conversion must be done in container compilation phase (or more generally when bootstrapping the configuration or services). We have plans in our roadmap to make this package runnable as a standalone app (without Symfony) so I wouldn't be against a proper reusable validation method somewhere.

Considering this, it would be great to add a `public static parseTimeout(null|int|float|string): ?int` method, which proceed to _user input_ validation I described upper (using `ctype_digit()` and `\DateInterval::createFromDateString()`). But this probably does not belong to this class. This method should also validate other constraints (more than 0 or null) even thought it would be duplicated work, because the validation also is required in the process builder as well (nothing prevent an API user from using it directly without using a validated user input).

This method could be written in a new `Validation` class within the `Helper` namespace.

This method can then be called when parsing configuration in the container extension, for validating the user input, and later in the future will be used in the standalone application bootstrap as well.

We'd very much like to keep an int here for the timeout, which will eventually be cast as float when passing it to Symfony process component when spawning it.

I'm OK with validating 0 as being considered "no timeout", but in order to be consistent with our code conventions, please consider setting `null` value to the class property when validating.

Reply:
i would suggest leaving 0 as "no timeout", otherwise handling the parameter could unnecessarily complicated together with the wish for a date interval from string to also work. Any programmer who has at least some experience with timeouts will assume that 0 means infinite, as it seems to be convention to me.

Reply:
> i would suggest leaving 0 as "no timeout",

This goes against about usual conventions of this project, but if you feel it to be better, I'm OK with an non-nullable `int` property whose `0` value means _no timeout_. This is a minor detail and I don't wish it to block this PR !